internal/core/command/v2/controller/http/command_test.go

```diff
@@ -183,3 +229,94 @@ func TestCommandsByDeviceName(t *testing.T) {
 		})
 	}
 }
+
+func TestIssueReadCommand(t *testing.T) {
+	var nonExistName = "nonExist"
+
+	expectedEventResponse := buildEventResponse()
+	expectedDeviceResponse := buildDeviceResponse()
+	expectedDeviceServiceResponse := buildDeviceServiceResponse()
+
+	dcMock := &mocks.DeviceClient{}
+	dcMock.On("DeviceByName", context.Background(), testDeviceName).Return(expectedDeviceResponse, nil)
+	dcMock.On("DeviceByName", context.Background(), nonExistName).Return(responseDTO.DeviceResponse{}, errors.NewCommonEdgeX(errors.KindEntityDoesNotExist, "fail to query device by name", nil))
+
+	dscMock := &mocks.DeviceServiceClient{}
+	dscMock.On("DeviceServiceByName", context.Background(), testDeviceServiceName).Return(expectedDeviceServiceResponse, nil)
+
+	dsccMock := &mocks.DeviceServiceCommandClient{}
+	dsccMock.On("GetCommand", context.Background(), testBaseAddress, testDeviceName, testCommandName, v2.ValueNo, v2.ValueYes).Return(expectedEventResponse, nil)
+	dsccMock.On("GetCommand", context.Background(), testBaseAddress, testDeviceName, nonExistName, v2.ValueNo, v2.ValueYes).Return(responseDTO.EventResponse{}, errors.NewCommonEdgeX(errors.KindContractInvalid, "fail to query device service by name", nil))
+
+	dic := NewMockDIC()
+	dic.Update(di.ServiceConstructorMap{
+		V2Container.MetadataDeviceClientName: func(get di.Get) interface{} { // add v2 API MetadataDeviceClient
+			return dcMock
+		},
+		V2Container.MetadataDeviceServiceClientName: func(get di.Get) interface{} { // add v2 API MetadataDeviceProfileClient
+			return dscMock
+		},
+		V2Container.DeviceServiceCommandClientName: func(get di.Get) interface{} { // add v2 API DeviceServiceCommandClient
+			return dsccMock
+		},
+	})
+	cc := NewCommandController(dic)
+	assert.NotNil(t, cc)
+
+	tests := []struct {
+		name               string
+		deviceName         string
+		commandName        string
+		pushEvent          string
+		returnEvent        string
+		errorExpected      bool
```

Comment:
You will need to add user query parameters and tests for them being passed to the Device Command.

Reply:
Added.